salt/modules/freebsd_update.py

```diff
@@ -0,0 +1,275 @@
+
+
+# Import python libs
+import logging
+
+# Import salt libs
+import salt
+import salt.utils.decorators as decorators
+from salt.exceptions import CommandNotFoundError
+
+
+log = logging.getLogger(__name__)
+
+# Define the module's virtual name
+__virtualname__ = 'freebsd-update'
+
+
+def __virtual__():
+    '''
+    .. versionadded:: 2016.3.4
+
+    Only work on FreeBSD RELEASEs >= 6.2, where freebsd-update was introduced.
+    '''
+    if __grains__['os'] != 'FreeBSD':
+        return (False, 'The freebsd_update execution module cannot be loaded: only available on FreeBSD systems.')
+    if float(__grains__['osrelease']) < 6.2:
+        return (False, 'freebsd_update is only available on FreeBSD versions >= 6.2-RELESE')
+    if 'release' not in __grains__['kernelrelease'].lower():
+        return (False, 'freebsd_update is only available on FreeBSD RELEASES')
+    return __virtualname__
+
+
+def _cmd(**kwargs):
+    '''
+    .. versionadded:: 2016.3.4
+
+    '''
+    update_cmd = salt.utils.which('freebsd-update')
+    if not update_cmd:
+        raise CommandNotFoundError('"freebsd-update" command not found')
+
+    params = []
+    if 'basedir' in kwargs:
+        params.append('-b {0}'.format(kwargs['basedir']))
+    if 'workdir' in kwargs:
+        params.append('-d {0}'.format(kwargs['workdir']))
+    if 'conffile' in kwargs:
+        params.append('-f {0}'.format(kwargs['conffile']))
+    if 'force' in kwargs:
+        params.append('-F')
+    if 'key' in kwargs:
+        params.append('-k {0}'.format(kwargs['key']))
+    if 'newrelease' in kwargs:
+        params.append('-r {0}'.format(kwargs['newrelease']))
+    if 'server' in kwargs:
+        params.append('-s {0}'.format(kwargs['server']))
+    if 'address' in kwargs:
+        params.append('-t {0}'.format(kwargs['address']))
+
+    return '{0} {1}'.format(update_cmd, ' '.join(params))
+
+
+def _wrapper(orig, pre='', post='', err_=None, run_args={}, **kwargs):
+    '''
+    Helper function that wraps the execution of freebsd-update command.
+
+    orig:
+        Originating function that called _wrapper().
+
+    pre:
+        String that will be prepended to freebsd-update command.
+
+    post:
+        String that will be appended to freebsd-update command.
+
+    err_:
+        Dictionary on which return codes and stout/stderr are copied.
+
+    run_args:
+        Arguments to be passed on cmd.run_all.
+
+    kwargs:
+        Parameters of freebsd-update command.
+    '''
+
+    cmd = _cmd(**kwargs)
+    res = __salt__['cmd.run_all']('{0} {1} {2} {3}'.format(pre, cmd, post, orig), **run_args)
+    if err_ is not None: # copy return values if asked to
+        for k, v in res.items():
+            err_[k] = v
+
+    if 'retcode' in res and res['retcode'] != 0:
+        log.error('Unable to run "{0} {1} {2} {3}". Error: {1}'.format(pre, cmd, post, orig, res['stderr']))
+        return res
+    return res['stdout']
+
+
+def fetch(pre='', post='', err_=None, run_args={}, **kwargs):
```

Comment:
I think (as explained in the pull request) that it could be safely be left as it reduces complexity and maintenance.